### DISCUSSION TITLE
proxy backend subdirectory

### DISCUSSION THREAD

**Discussion by asheetal** [open-webui/open-webui#6650] [General]
open-webui currently only works at the root directory, for example www.mywebsite.com. It does not work on server that already has a root content. In other words www.mywebsite.com/chatgpt does not work where the root is served by nginx.

Steps to reproduce

Server 1: Root server www.mywebsite.com running nginx and a wordpress website.

Server 2: open-webui server running open-webui and ollama.

**Comment by npapke** [comment]
I am also running into this issue and I too believe that supporting reverse proxies with a subpath would be a useful feature. 

## Current Behavior
I have tried adding a rewrite rule to the nginx configuration.  Something like
``` 
 rewrite ^/openwebui/?(.*)$ /$1 break;
```

This allows requests to be processed.  However, responses that include paths are missing the "openwebui" path component.

Static asset URLs in the HTML response are missing the base path:
```html
   <link rel="modulepreload" href="/_app/immutable/entry/start.Bqm64-9x.js">
   <link rel="icon" href="/favicon.png" />
```

The SvelteKit base configuration in the HTML shows an empty base path:
```html
__sveltekit_129w8ty = {
    base: ""
};
```

## Desired Behavior
Ability to set the base context path via an environment variable or similar.

## Environment
OCI image from `ghcr.io/open-webui/open-webui:main`

**Comment by Jatin794** [comment]
I tried both docker and python deployments and this isn't currently possible. Though maybe in the future we'll be able to edit that using environment configs

**Comment by rgaricano** [comment]
In nginx you can setup a new location for your openwebui directory, /chatgpt  ( in your case) with a proxy_pass to your openwebui server

eg:
```
server {

    server_name www.mywebsite.com);
    listen 80;
 #location1: your webroot directory pointing to your wordpress config 

    location /
	{
       ...
      }
#location2: your web chatgpt directory pointing to your openwebui server (localhst:8080 in this example)

    location /chatgpt
      {
       proxy_set_header Host $host;
       proxy_pass http://localhost:8080;
       proxy_set_header Accept-Encoding "";
       proxy_set_header X-Real-IP $remote_addr;
       proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
       proxy_http_version 1.1;
       proxy_set_header Upgrade $http_upgrade;
       proxy_set_header Connection "upgrade";
       proxy_set_header X-Forwarded-Scheme $scheme;
       proxy_set_header X-Forwarded-Proto $scheme;
       keepalive_timeout 3600;
       proxy_read_timeout 3600;
       proxy_connect_timeout 3600;
       proxy_send_timeout 3600;
	}

}

```

**Reply by Jatin794** [reply]
Nope. Added the few parameters I didn't have before from here, but that hasn't changed the error at all. None of the assets load, as confirmed by network inspector.

**Comment by extreme-sysadminning** [comment]
Same problem, except that i use Apache as a proxy. I started doing sed style substitutions to the webtraffic between open-webui and Apache, but that didn't solve much.

The problem has been raised elsewhere:
https://github.com/open-webui/open-webui/issues/1162

A patch has been made a few weeks ago, but it was rejected:
https://github.com/hsmallbone/open-webui/compare/open-webui%3Aopen-webui%3Amain...relative-urls

It appears the only solution is to have open-webui in the web root for now, which in my case means buying a DNS hostname and a SSL certificate.

**Comment by klyoungblood** [comment]
I'm also unable open-webui on a sub-directory in my deployment with nginx, with static asserts failing to load. Any update from the maintainers?

**Comment by extreme-sysadminning** [comment]
My understanding is this is a WONTFIX at this time. I opted to use another Apache VHOST and called it a day.

**Reply by brookst** [reply]
That's not possible for me since I don't have a wildcard cert. MR #2148 is where the WONTFIX is stated. I tried doing the same as yourself with Apache Substitute rules, but likewise got stuck untangling all the references from javascript. I'd appreciate getting this feature accepted in to open-webui.

**Comment by ronsmith** [comment]
I can't believe this is still an issue over a year later.

**Reply by Classic298** [reply]
What's the issue?
Are you meaning something else? All i can find here is a feature request.

**Reply by ronsmith** [reply]
If you read above, fixes were posted and rejected, and issues were marked WONTFIX, but the problem still exists.

**Reply by Classic298** [reply]
I only see a closed feature request with comment that this feature won't be added.

I don't see a proposed fix with wontfix

**Reply by ronsmith** [reply]
https://github.com/open-webui/open-webui/pull/2148#issuecomment-2558116333
"Relative path support is not something we intend to officially support at this time. "
Closed with unmerged commits.

**Reply by Classic298** [reply]
Ok. As that's a feature request that tim said won't be added but i don't see an issue that's not fixed anywhere.

**Reply by ronsmith** [reply]
You're splitting hairs. This discussion was converted from an issue. Convert it back, and you've got your issue.
